src/lib/packager/clients/pkg_finish.rb

```
@@ -0,0 +1,192 @@
+      remove_old_sources(current_repos)
+
+      # Sync sources
+      sync_target_sources
+
+      nil
+    end
+
+    # Disable given repositories if needed
+    #
+    # Given a local repository:
+    #
+    # * if all products it contains are available through another repository,
+    #   then it will be disabled;
+    # * if some product contained is not available through another
+    #   repository, then it will be left untouched.
+    #
+    # @return [Array<Packages::Repository>] List of disabled repositories
+    def disable_local_repos
+      candidates_repos, other_repos = *::Packages::Repository.enabled.partition(&:local?)
+      products = other_repos.map(&:products).flatten.uniq
+      candidates_repos.each_with_object([]) do |repo, disabled|
+        uncovered = repo.products.reject { |p| products.include?(p) }
+        if uncovered.empty?
+          log.info("Repo #{repo.repo_id} will be disabled because products are present "\
+            "in other repositories")
+          repo.disable!
+          disabled << repo
+        else
+          log.info("Repo #{repo.repo_id} cannot be disabled because these products " \
+                   "are not available through other repos: #{uncovered.map(&:name)}")
+        end
+      end
+    end
+
+    # Backup sources
+    #
+    # @param source [String] Path of sources to backup
+    # @param target [String] Directory to store backup
+    # @return [String] Name of the backup archive (locate in the given target directory)
+    def backup_old_sources(source, target)
+      archive_name = "repos_#{Time.now.strftime(BACKUP_TIMESTAMP_FORMAT)}.tgz"
+      compress_cmd = format(TAR_CMD,
+        target: String.Quote(target),
+        archive: archive_name,
+        source: String.Quote(source))
+      cmd = SCR.Execute(path(".target.bash_output"), compress_cmd)
+      if !cmd["exit"].zero?
+        log.error("Unable to backup current repos; Command >#{compress_cmd}< returned: #{cmd}")
```

If it fails then the returned value is the result of `log.error` which is wrong. It should either throw an exception or return `nil` on error.